extensions/markdown/src/extension.ts

```diff
@@ -17,31 +17,66 @@ interface IPackageInfo {
 
 var telemetryReporter: TelemetryReporter | null;
 
+interface IToken {
+	type: string;
+	map: [number, number];
+}
+
+interface MarkdownIt {
+	render(text: string): string;
+	parse(text: string): IToken[];
+}
+
+function createMarkdownIt(): MarkdownIt {
+	const hljs = require('highlight.js');
+	const mdnh = require('markdown-it-named-headers');
+	const md = require('markdown-it')({
+		html: true,
+		highlight: (str: string, lang: string) => {
+			if (lang && hljs.getLanguage(lang)) {
+				try {
+					return `<pre class="hljs"><code><div>${hljs.highlight(lang, str, true).value}</div></code></pre>`;
+				} catch (error) { }
+			}
+			return `<pre class="hljs"><code><div>${md.utils.escapeHtml(str)}</div></code></pre>`;
+		}
+	}).use(mdnh, {});
+
+	return md;
+}
+
+const FrontMatterRegex = /^---\s*(.|\s)*?---\s*/;
+
 export function activate(context: vscode.ExtensionContext) {
 
 	let packageInfo = getPackageInfo(context);
 	telemetryReporter = packageInfo && new TelemetryReporter(packageInfo.name, packageInfo.version, packageInfo.aiKey);
 
-	let provider = new MDDocumentContentProvider(context);
-	let registration = vscode.workspace.registerTextDocumentContentProvider('markdown', provider);
+	const markdownIt = createMarkdownIt();
```

Comment:
ping @mjbvz. Is requiring those 3 node modules really something worth delaying? Did we find performance issues with it?